components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * you may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.apimgt.impl.dao;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.dto.CertificateMetadataDTO;
+import org.wso2.carbon.apimgt.impl.certificatemgt.CertificateManagementException;
+import org.wso2.carbon.apimgt.impl.utils.APIMgtDBUtil;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class handles data base transactions related to the server certificate management.
+ * Here the certificate information will be,
+ * Alias
+ * Server Endpoint
+ * Tenant Id
+ * <p>
+ * The alias and server ep has one-to-one mapping, which there could be one certificate for a particular endpoint.
+ * A tenant could have multiple alias-ep entries.
+ */
+public class CertificateMgtDAO {
+
+    private static final String CERTIFICATE_TABLE_NAME = "AM_CERTIFICATE_METADATA";
+    private static Log log = LogFactory.getLog(CertificateMgtDAO.class);
+    private static CertificateMgtDAO certificateMgtDAO = null;
+    private static boolean initialAutoCommit = false;
+
+    /**
+     * Private constructor
+     */
+    private CertificateMgtDAO() {
+    }
+
+    /**
+     * Returns an instance of CertificateMgtDao.
+     */
+    public static CertificateMgtDAO getInstance() {
+        if (certificateMgtDAO == null) {
+            certificateMgtDAO = new CertificateMgtDAO();
+        }
+        return certificateMgtDAO;
+    }
+
+    /**
+     * Checks whether the certificate management table exists in the data base.
+     *
+     * @return : True if exists, false otherwise.
+     */
+    public boolean isTableExists() {
+        boolean isExists = false;
+        Connection connection = null;
+        DatabaseMetaData databaseMetaData = null;
+
+        try {
+            connection = APIMgtDBUtil.getConnection();
```

Comment:
use java8 syntac here

Reply:
In apim-6.1.x we could not use java8 syntax.

Reply:
Got it we need to close ResultSet as well please do needful

Reply:
Closed the result set.